pkg/controller/cloneset/sync/cloneset_scale.go

```diff
@@ -60,6 +64,19 @@ func (r *realControl) Scale(
 		return false, nil
 	}
 
+	// If cloneset doesn't want to reuse pvc, clean up
+	// the existing pvc first. Then it looks like the pod
+	// is deleted by controller, new pod can be created.
+	if updateCS.Spec.ScaleStrategy.DisablePVCReuse {
+		uselessPVCs := getUselessPVCs(pods, pvcs)
+		if len(uselessPVCs) > 0 {
+			klog.V(3).Infof("Begin to clean up cloneset %s useless PVCs", controllerKey)
+			if modified, err := r.cleanupPVCs(updateCS, uselessPVCs); err != nil || modified {
+				return modified, err
```

Comment:
if pvc is modified, why return? I think the controller can keep reconciling.

Reply:
In the existing logic, if there is a resource change, the controller will return.

Reply:
I suppose the "existing logic" in the comment is this code block,

'''
	podsSpecifiedToDelete, podsInPreDelete, numToDelete := getPlannedDeletedPods(updateCS, pods)
	if modified, err := r.managePreparingDelete(updateCS, pods, podsInPreDelete, numToDelete); err != nil || modified {
		return modified, err
	}
'''
It re-enqueues not because it has mark some pods to delete, but because preparing-to-delete pods has to be processed by other operators first. 

In general, if one try to modify multiple underlying cloneset resources, pods, pvc etc) , modification can be made in one reconcile loop, so as to speed the reconcilation

Reply:
Thank you for your explanation, this can indeed be optimized.